sim_pipeline/galaxy_galaxy_lens.py

```diff
@@ -415,13 +419,97 @@ def lenstronomy_kwargs(self, band=None):
+        (
+            lens_light_model_list,
+            kwargs_lens_light,
+        ) = self.deflector_light_model_lenstronomy(band=band)
+
         kwargs_model = {
-            "lens_light_model_list": ["SERSIC_ELLIPSE"],
-            "lens_model_list": lens_model_list,
+            "lens_light_model_list": lens_light_model_list,
+            "lens_model_list": lens_mass_model_list,
+        }
+
+        sources, sources_kwargs = self.source_light_model_lenstronomy(band=band)
+        # ensure that only the models that exist are getting added to kwargs_model
+        for k in sources.keys():
+            kwargs_model[k] = sources[k]
+
+        kwargs_source = sources_kwargs["kwargs_source"]
+        kwargs_ps = sources_kwargs["kwargs_ps"]
+
+        kwargs_params = {
+            "kwargs_lens": kwargs_lens,
+            "kwargs_source": kwargs_source,
+            "kwargs_lens_light": kwargs_lens_light,
+            "kwargs_ps": kwargs_ps,
         }
 
-        center_lens, center_source = self.lens_position, self.source_position
+        return kwargs_model, kwargs_params
+
+    def deflector_mass_model_lenstronomy(self):
+        """Returns lens model instance and parameters in lenstronomy conventions.
+
+        :return: lens_model_list, kwargs_lens
+        """
+        lens_mass_model_list = ["EPL", "SHEAR", "CONVERGENCE"]
+        theta_E = self.einstein_radius
+        e1_light_lens, e2_light_lens, e1_mass, e2_mass = self.deflector_ellipticity()
+        center_lens = self.deflector_position
+        gamma1, gamma2, kappa_ext = self.los_linear_distortions()
+        kwargs_lens = [
+            {
+                "theta_E": theta_E,
+                "gamma": 2,
+                "e1": e1_mass,
+                "e2": e2_mass,
+                "center_x": center_lens[0],
+                "center_y": center_lens[1],
+            },
+            {"gamma1": gamma1, "gamma2": gamma2, "ra_0": 0, "dec_0": 0},
+            {"kappa": kappa_ext, "ra_0": 0, "dec_0": 0},
+        ]
+
+        return lens_mass_model_list, kwargs_lens
+
+    def deflector_light_model_lenstronomy(self, band=None):
+        """Returns lens model instance and parameters in lenstronomy conventions.
+
+        :return: lens_light_model_list, kwargs_lens_light
+        """
+        lens_light_model_list = ["SERSIC_ELLIPSE"]
+        center_lens = self.deflector_position
+        e1_light_lens, e2_light_lens, e1_mass, e2_mass = self.deflector_ellipticity()
+        size_lens_arcsec = (
+            self._deflector_dict["angular_size"] / constants.arcsec
+        )  # convert radian to arc seconds
+
+        if band is None:
+            mag_lens = 1
+        else:
+            mag_lens = self.deflector_magnitude(band)
+        kwargs_lens_light = [
+            {
+                "magnitude": mag_lens,
+                "R_sersic": size_lens_arcsec,
+                "n_sersic": float(self._source_dict["n_sersic"]),
```

Comment:
I don't know why this is new code, but we fixed a bug here switching _source_dict to _deflector_dict

Reply:
Will address these comments now and push again!